### PR TITLE
Datepicker for start and end time querying

### PR DESCRIPTION
**Pull request by alahwa**

### CODE REVIEW

File: ui/src/app/shared/common.ts [hunk 1]
```diff
@@ -10,15 +10,19 @@ export enum JobStatusImage {
   Aborted = <any> 'https://www.gstatic.com/images/icons/material/system/1x/report_problem_grey600_24dp.png'
 }
 
+export const start = 'start';
```
Comment (calbach) [inline ui/src/app/shared/common.ts]: nit: I might call these `startCol`, `endCol`

File: ui/src/app/shared/header/header.component.html [hunk 2]
```diff
@@ -10,14 +10,26 @@
              src="https://www.gstatic.com/images/icons/material/system/1x/cancel_grey600_18dp.png">
       </mat-chip>
       <mat-menu #menu="matMenu" [overlapTrigger]="false" class="wide-menu">
-        <mat-form-field class="query-input">
+        <mat-form-field *ngIf="isFreeText()" class="query-input">
```
Comment (calbach) [inline ui/src/app/shared/header/header.component.html]: opt: `*ngSwitch` might be nice here and would ensure mutual exclusion for these being displayed

File: ui/src/app/shared/header/header.component.spec.ts [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
 } from "@angular/material";
 import {FormsModule, ReactiveFormsModule} from "@angular/forms";
 import {RouterTestingModule} from "@angular/router/testing";
 import {BrowserAnimationsModule} from "@angular/platform-browser/animations";
 
+import {HeaderComponent} from "./header.component";
+import {start} from "../common";
+
 
 describe('HeaderComponent', () => {
```
Comment (calbach) [inline ui/src/app/shared/header/header.component.spec.ts]: Unit test for date picking?

File: ui/src/app/shared/common.ts (hunk 1 above)
Comment (alahwa) [reply ui/src/app/shared/common.ts]: Done

File: ui/src/app/shared/header/header.component.spec.ts (hunk 3 above)
Comment (alahwa) [reply ui/src/app/shared/header/header.component.spec.ts]: Added

**Pull request merged**